src/display.jl

```diff
@@ -75,7 +75,22 @@ end
 plot(args...; kwargs...) = SyncPlot(Plot(args...; kwargs...))
 
 ## API methods for SyncPlot
-for f in [:restyle!, :relayout!, :addtraces!, :deletetraces!, :movetraces!,
+function restyle!(sp::SyncPlot, args...; kwargs...)
+    diffs = restyle!(sp.plot, args...; kwargs...)
+    # each trace in the plot returns a diff, so we have to apply them
+    # separately. Maybe at some point we can do some clever merging
+    for (i, diff) in enumerate(diffs)
```

Comment:
Hmm, I'm looking closer at this and how can we be sure that the `i` from enumerate is the actual `i` the user chose when he was restyling a specific trace?

I have some ideas for how to do the merging of diffs on the Julia side, but want to make sure I understand everything you have right now first